feat: device update handling in main app

TejasvCypherock opened the pull request.

TejasvOnly committed:
feat: device update handling in main app

TejasvOnly committed:
feat: handle incompatible device case

TejasvOnly committed:
Merge remote-tracking branch 'origin/develop' into feat/device-update-main-app

irshadCypherock commented on packages/cysync-core/src/dialogs/AppUpdateDialog.tsx:
```diff
@@ -0,0 +1,147 @@
+  ProgressDialog,
+  SuccessDialog,
+} from '@cypherock/cysync-ui';
+import { BlurOverlay } from '@cypherock/cysync-ui/src';
```
Can merge both of these imports

TejasvOnly committed:
feat: handle device cases in flows

TejasvOnly committed:
chore: add changeset

TejasvOnly committed:
Merge remote-tracking branch 'origin/develop' into feat/device-update-main-app

TejasvOnly committed:
fix: review changes

- dont show device update bar in bootloader mode
- show error on disconnect while device is updating

irshadCypherock commented on packages/cysync-core/src/dialogs/ConnectionErrorDialog.tsx:
```diff
@@ -0,0 +1,78 @@
+  useAppSelector,
+} from '..';
+
+export const ConnectionErrorDialog: FC = () => {
```
Unused component

irshadCypherock commented on packages/cysync-core/src/hooks/fetchDeviceHandlingState.ts:
```diff
@@ -0,0 +1,61 @@
+  UNKNOWN_ERROR,
+}
+
+export const fetchDeviceHandlingState = () => {
```
We can move this to useDevice context

irshadCypherock commented on packages/cysync-core/src/hooks/useAppUpdate.ts:
```diff
@@ -68,8 +69,8 @@ export const useAppUpdate = ({ shouldInstallAfterUpdate }: AppUpdateProps) => {
       setInternalState(InternalAppUpdateState.Installing);
       await autoUpdater.installUpdate();
       setAppUpdateState(AppUpdateState.Successful);
```
I think this should be conditional

TejasvOnly committed:
chore: review changes

TejasvOnly committed:
Merge remote-tracking branch 'origin/develop' into feat/device-update-main-app

TejasvOnly committed:
Merge remote-tracking branch 'origin/develop' into feat/device-update-main-app

TejasvOnly committed:
Merge remote-tracking branch 'origin/develop' into feat/device-update-main-app

TejasvOnly committed:
fix: app update fixes
